### PR TITLE
Validate rules docs when running unit tests

### LINKED ISSUE #1734: Generate Rules.md automatically
It would be great if we could automate the generation of `Rules.md` even more, so we wouldn't even need to run its scheme. Maybe somehow running it together with unit tests?

Another option would be generating whenever something is merged in master.

### PR DESCRIPTION
**Pull request by marcelofabri**
Fix #1734

I'm not sure if this deserves a changelog entry.

### CODE REVIEW

File: Tests/SwiftLintFrameworkTests/DocumentationTests.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    func testRulesDocumentationIsUpdated() throws {
+        let docsPath = "\(projectRoot)/Rules.md"
+        let existingDocs = try String(contentsOfFile: docsPath)
+        let updatedDocs = masterRuleList.generateDocumentation()
+
+        XCTAssertEqual(existingDocs, updatedDocs)
+
+        let overwrite = false // set this to true to overwrite existing docs with the generated ones
+        if existingDocs != updatedDocs && overwrite {
+            try updatedDocs.data(using: .utf8)?.write(to: URL(fileURLWithPath: docsPath))
```
Comment (marcelofabri) [inline Tests/SwiftLintFrameworkTests/DocumentationTests.swift]: I copied this approach from SourceKitten but I wonder if we shouldn't just always update the docs 🙃

File: Source/SwiftLintFramework/Models/RuleList+Documentation.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+//  SwiftLint
+//
+//  Created by Marcelo Fabri on 8/24/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+
+extension RuleList {
+    public func generateDocumentation() -> String {
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Models/RuleList+Documentation.swift]: I'd prefer to keep it outside `SwiftLintFramework`, but I think it's a reasonable compromise.

File: Tests/SwiftLintFrameworkTests/DocumentationTests.swift (hunk 1 above)
Comment (jpsim) [reply Tests/SwiftLintFrameworkTests/DocumentationTests.swift]: yes. the reason it's parameterized in SourceKitten is that its output is non-deterministic.

**Comment by jpsim** [comment]
Finally re-enabled this test in af14fb25f08b5c63c209f24c966b4e2d4f3c9e7c

**Pull request merged**